### ISSUE TITLE
Links are misconfigured

### ISSUE THREAD

**Issue by stefanschramek** [anonaddy/anonaddy#36]
Hi,
From time to time I have issues with links sent through AnonAddy.
The links are invalid when I click them, and I saw that some of the dots in the URL are doubled, e.g.
`https://www.domain..com/email?e=JXkKl49H8xTHLpLq3cRl4RFfogdRojsxthdswwWFvSJtZZnSb9eNnNQ5dK7iprp3`

Thanks
Stefan

**Comment by willbrowningme** [comment]
Hmm, are you using GPG encryption?

Is this for the plain text version of the email or html?

Thanks!

**Comment by stefanschramek** [comment]
As far as I remember it‘s happening in plain text emails!
No GPG encryption used!

Thanks
Stefan

**Comment by stefanschramek** [comment]
I still have this issue from time to time!
Is there any update?
I have one from today, which I could forward to you if it helps debugging!

Thanks!

**Comment by willbrowningme** [comment]
I haven't been able to replicate this issue. Please could you send me an email with further details and perhaps the email attached too (with any sensitive information removed).

**Comment by willbrowningme** [comment]
Does anyone have any ideas about this issue? It seems to happen very occasionally with certain URLs, perhaps it is related to the base64 encoding then decoding.

**Comment by jlopp** [comment]
I will note that when I was investigating this issue for a user who reported our email links as broken, the particular spot in the HTML portion of the email where the URL got screwed up ALSO happened to coincide with a newline. That may or may not be a root cause. Example section:

```
href=3D"https://api.keys.=
casa/verifyemail/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.=
eyJpZCI6ImI2ODI0NDdkLTA4ZWItNDA0MS04ZjU4LTU2Njg5MjMwZGM0MCIsImp0aSI6IjI2NmV=
iMWRhLWJmNWQtNDYxMC05NzMxLTQ5NjdkY2ZmNDc0OCIsInR5cGUiOiJ2ZW0iLCJpc3MiOiIwNG=
ZlOTg0NS02MGVlLTUxYzgtOTVhYi1lNTgyOGUwNDVhZTYiLCJleHAiOjE1OTMyMDUxNDd9.=
8gTQ3AP21y8BsInYP0uPqoyZ43bDVqSkITrXuykeBUE"
```

**Comment by willbrowningme** [comment]
@jlopp Thanks a lot, I still can't seem to reproduce this locally or in production. Do you happen to have an example of the full raw .eml file of the email where this issue occurred?

**Comment by jlopp** [comment]
I'm having trouble reproducing it myself; will email you the info I have.

**Comment by vlenzr** [comment]
Got the same problem. Tried to change a mail address for an online service. Therefore I had to confirm my new mail address via a send link. I nearly went crazy as I tried it dozens of times (each time with a new link). The customer service pointed out to me the additional point in the link. I can forward the mail/ link if desired.

**Comment by willbrowningme** [comment]
@vlenzr Yes please could you let me know the service so I can test this out myself. You can email it to me if you like.

**Comment by vlenzr** [comment]
@willbrowningme it's www.ebay-kleinanzeigen.de. Will forward you the mail.

**Comment by vlenzr** [comment]
@willbrowningme it's www.ebay-kleinanzeigen.de. I can't find an english version. If you need any help please let me know. In addition I will forward you the mail.

**Comment by erik-h** [comment]
Just wanted to add that I've also experienced this issue a couple of times. If it helps I can find one of my affected emails and forward it along as well.

**Comment by cmbail** [comment]
I'm seeing this regularly. Just sent @willbrowningme the eml for the most recent one. I reviewed other messages of the same type from the same sender and the other links I tested were fine. I tried searching my email for “..com” and unfortunately that also returns results for “.com”. In Outlook, Gmail (web), and X1 Search. Let me know if you have any other tips for surfacing these?!

**Comment by willbrowningme** [comment]
I believe I have just pushed out a fix for this issue in release v0.2.12, specificully this [commit](https://github.com/anonaddy/anonaddy/commit/8b65ec58bef5211a909a539ad17fb67fbab2f839).

I found that the bug had been widely discussed on the SwiftMailer repo and numerous issues had been opened for it, the latest being - https://github.com/swiftmailer/swiftmailer/issues/805

The last user's comment about escaping the periods for the quoted printable encoding seems to be working so far.

If anybody still sees this double dot issue occurring then please let me know.
